src/ocean/util/UnixSocketCommandInterface.d

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+
+    Utility for handling requests via a unix socket.
+
+    Commands can be registered to the extension and when the command is received
+    the provided delegate will be called. Any provided arguments will be split
+    by the space character and provided as an array.
+
+    Copyright:
+        Copyright (c) 2009-2018 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.util.UnixSocketCommandInterface;
+
+import ocean.transition;
+import ocean.core.Verify;
+
+import ocean.core.Enforce;
+import ocean.io.select.EpollSelectDispatcher;
+
+/// ditto
+public class UnixSocketCommandInterface
```

Comment:
Any reason for this to be a class?

Reply:
No, good catch :+1:

Reply:
Sad, the bit with the connection handlers needs to be a class, since the `UnixListener` takes a reference of the current instance. Making this a struct would copy it and `addHandler`/`removeHandler` would no longer reflect on the actual connection handler.